[Doc]: Unclear how to compile `c_internals` in code space

### Documentation Link

https://matplotlib.org/devdocs/devel/development_setup.html#create-a-dedicated-environment

### Problem

Actually, I run into a little bit into trouble to get the `c_internals`  compiled. After reading the docs, it sounds like that the _Codespace_ environment is set up in way that out of the box it can be used. So I am wondering if there can be:


1. provide further instructions how to compile the  `c_internals`  
2. or update the `.devcontainer/setup.sh`

I know about this section [create-a-dedicated-environment](https://matplotlib.org/devdocs/devel/development_setup.html#create-a-dedicated-environment), but I haven't managed to get it running in _Codespace_. 

> Other users might face similar struggles. If there's an option in the shell to autocompile during the CodeSpace setup, it would make this environment very convenient.

Actually, I was planning to address this blind spot in the coverage rate of the **PR** #29520 , but I wasn't able to get `pytest` running.




```terminal
ERROR lib/mpl_toolkits/axes_grid1/tests - ImportError: cannot import name '_c_internal_utils' from 'matplotlib' (/workspaces/matplotlib/lib/matplotlib/__init__.py)
ERROR lib/mpl_toolkits/axisartist/tests - ImportError: cannot import name '_c_internal_utils' from 'matplotlib' (/workspaces/matplotlib/lib/matplotlib/__init__.py)
ERROR lib/mpl_toolkits/mplot3d/tests - ImportError: cannot import name '_c_internal_utils' from 'matplotlib' (/workspaces/matplotlib/lib/matplotlib/__init__.py)
```

Further Sources:

* https://discourse.matplotlib.org/t/cannot-import-name-c-internal-utils-from-matplotlib/22644
* https://github.com/matplotlib/matplotlib/pull/29520/files#annotation_31349915228

### Suggested improvement

* Include procedure for compiling `c_internal` into [.devcontainer/setup.sh](https://github.com/matplotlib/matplotlib/blob/main/.devcontainer/setup.sh)
* Provide some details about how to `gcc` via conda.

@rcomer Thank you very much! for the quick answer ... unfortunately, after activating the enviroment. I really get the error.


<img width="680" alt="Image" src="https://github.com/user-attachments/assets/7fc50216-ac0b-47be-bad3-70b1fb1c4e4f" />

I also have tested via `conda run` or `python -m pytest`

**from the log-file**
```terminal
============================= test session starts ==============================
platform linux -- Python 3.12.8, pytest-8.3.4, pluggy-1.5.0 -- /home/codespace/micromamba/envs/mpl-dev/bin/python3.12
cachedir: .pytest_cache
rootdir: /workspaces/matplotlib
configfile: pyproject.toml
testpaths: lib
plugins: anyio-4.7.0, cov-6.0.0, timeout-2.3.1, rerunfailures-15.0, xdist-3.6.1
collecting ... collected 168 items / 86 errors / 1 skipped

==================================== ERRORS ====================================
___________________ ERROR collecting lib/matplotlib/testing ____________________
<frozen importlib._bootstrap_external>:999: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
lib/matplotlib/__init__.py:161: in <module>
    from . import _api, _version, cbook, _docstring, rcsetup
lib/matplotlib/cbook.py:32: in <module>
    from matplotlib import _api, _c_internal_utils
E   ImportError: cannot import name '_c_internal_utils' from 'matplotlib' (/workspaces/matplotlib/lib/matplotlib/__init__.py)
______________ ERROR collecting lib/matplotlib/tests/test_afm.py _______________
ImportError while importing test module '/workspaces/matplotlib/lib/matplotlib/tests/test_afm.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/matplotlib/tests/test_afm.py:6: in <module>
    from matplotlib import font_manager as fm
lib/matplotlib/font_manager.py:48: in <module>
    from matplotlib import _api, _afm, cbook, ft2font
lib/matplotlib/cbook.py:32: in <module>
    from matplotlib import _api, _c_internal_utils
E   ImportError: cannot import name '_c_internal_utils' from 'matplotlib' (/workspaces/matplotlib/lib/matplotlib/__init__.py)
______________ ERROR collecting lib/matplotlib/tests/test_agg.py _______________
```

if there is something missing a like post-compiling comando. Happy to know

--- a/doc/devel/development_setup.rst
+++ b/doc/devel/development_setup.rst
@@ -34,6 +34,22 @@ repository, you should first fork this repository by *clicking* the
 This creates a copy of the code under your account on the GitHub server. See `the GitHub
 documentation <https://docs.github.com/get-started/quickstart/fork-a-repo>`__ for more details.
 
+Decide whether to work locally or in GitHub Codespaces
+======================================================
+
+You can either work locally on your machine, or online in
+`GitHub Codespaces <github-codespaces_>`_, a cloud-based in-browser development
+environment.  If you are making a one-off, relatively simple change then
+working in GitHub Codespaces can be a good option because most of the setting
+up is done for you and you can skip the next few sections!  If you are making
+extensive or frequent contributions to Matplotlib then it is probably worth
+taking the time to set up on your local machine: As well as having the
+convenience of your local familiar tools, you will not need to worry about
+Codespace's monthly usage limits.
+
+If you want to use Codespaces, skip to :ref:`development-codespaces`,
+otherwise, continue with the next section.
+
 Retrieve the latest version of the code
 =======================================
 
@@ -106,8 +122,8 @@ code, as described in :ref:`development-workflow`.
 
 .. _dev-environment:
 
-Create a dedicated environment
-==============================
+Create a dedicated python environment
+=====================================
 You should set up a dedicated environment to decouple your Matplotlib
 development from other Python and Matplotlib installations on your system.
 
@@ -158,49 +174,6 @@ setup.
 
       Remember to activate the environment whenever you start working on Matplotlib!
 
-   .. tab-item:: :octicon:`codespaces` GitHub Codespaces
-
-      `GitHub Codespaces <https://docs.github.com/codespaces>`_ is a cloud-based
-      in-browser development environment that comes with the appropriate setup to
-      contribute to Matplotlib.
-
-      #. Open codespaces on your fork by clicking on the green :octicon:`code` ``Code``
-         button on the GitHub web interface and selecting the ``Codespaces`` tab.
-
-      #. Next, click on "Open codespaces on <your branch name>". You will be
-         able to change branches later, so you can select the default
-         ``main`` branch.
-
-      #. After the codespace is created, you will be taken to a new browser
-         tab where you can use the terminal to activate a pre-defined conda
-         environment called ``mpl-dev``::
-
-         conda activate mpl-dev
-
-      Remember to activate the *mpl-dev* environment whenever you start working on
-      Matplotlib.
-
-      If you need to open a GUI window with Matplotlib output on Codespaces, our
-      configuration includes a `light-weight Fluxbox-based desktop
-      <https://github.com/devcontainers/features/tree/main/src/desktop-lite>`_.
-      You can use it by connecting to this desktop via your web browser. To do this:
-
-      #. Press ``F1`` or ``Ctrl/Cmd+Shift+P`` and select
-         ``Ports: Focus on Ports View`` in the VSCode session to bring it into
-         focus. Open the ports view in your tool, select the ``noVNC`` port, and
-         click the Globe icon.
-      #. In the browser that appears, click the Connect button and enter the desktop
-         password (``vscode`` by default).
-
-      Check the `GitHub instructions
-      <https://github.com/devcontainers/features/tree/main/src/desktop-lite#connecting-to-the-desktop>`_
-      for more details on connecting to the desktop.
-
-      If you also built the documentation pages, you can view them using Codespaces.
-      Use the "Extensions" icon in the activity bar to install the "Live Server"
-      extension. Locate the ``doc/build/html`` folder in the Explorer, right click
-      the file you want to open and select "Open with Live Server."
-
 
 Install external dependencies
 =============================
@@ -216,6 +189,54 @@ Additionally, the following non-Python dependencies must also be installed local
 
 For a full list of dependencies, see :ref:`dependencies`. External dependencies do not
 need to be installed when working in codespaces.
+
+.. _development-codespaces:
+
+:octicon:`codespaces` Create a GitHub Codespace
+===============================================
+
+`GitHub Codespaces <github-codespaces_>`_ is a cloud-based
+in-browser development environment that comes with the appropriate setup to
+contribute to Matplotlib.
+
+#. Open codespaces on your fork by clicking on the green :octicon:`code` ``Code``
+   button on the GitHub web interface and selecting the ``Codespaces`` tab.
+
+#. Next, click on "Open codespaces on <your branch name>". You will be
+   able to change branches later, so you can select the default
+   ``main`` branch.
+
+#. After the codespace is created, you will be taken to a new browser
+   tab where you can use the terminal to activate a pre-defined conda
+   environment called ``mpl-dev``::
+
+      conda activate mpl-dev
+
+Remember to activate the *mpl-dev* environment whenever you start working on
+Matplotlib.
+
+If you need to open a GUI window with Matplotlib output on Codespaces, our
+configuration includes a `light-weight Fluxbox-based desktop
+<https://github.com/devcontainers/features/tree/main/src/desktop-lite>`_.
+You can use it by connecting to this desktop via your web browser. To do this:
+
+#. Press ``F1`` or ``Ctrl/Cmd+Shift+P`` and select
+   ``Ports: Focus on Ports View`` in the VSCode session to bring it into
+   focus. Open the ports view in your tool, select the ``noVNC`` port, and
+   click the Globe icon.
+#. In the browser that appears, click the Connect button and enter the desktop
+   password (``vscode`` by default).
+
+Check the `GitHub instructions
+<https://github.com/devcontainers/features/tree/main/src/desktop-lite#connecting-to-the-desktop>`_
+for more details on connecting to the desktop.
+
+If you also built the documentation pages, you can view them using Codespaces.
+Use the "Extensions" icon in the activity bar to install the "Live Server"
+extension. Locate the ``doc/build/html`` folder in the Explorer, right click
+the file you want to open and select "Open with Live Server."
+
+.. _`github-codespaces`: https://docs.github.com/codespaces
 
 .. _development-install:
 

